sdk/core/azure-core-amqp/CMakeLists.txt

```diff
@@ -126,10 +127,10 @@ set(AZURE_CORE_AMQP_SOURCE
     src/private/package_version.hpp
 )
 
-add_library(azure-core-amqp ${AZURE_CORE_AMQP_SOURCE} ${AZURE_CORE_AMQP_HEADER})
+add_library(azure-core-amqp ${AZURE_CORE_AMQP_SOURCE} ${AZURE_CORE_AMQP_HEADER} $<TARGET_OBJECTS:uamqp>)
 
 if (VENDOR_UAMQP)
-target_compile_definitions(azure-core-amqp PRIVATE _CRT_SECURE_NO_WARNINGS)
+target_compile_definitions(uamqp PRIVATE _CRT_SECURE_NO_WARNINGS)
```

Comment:
We need to update the azure-core-amqp definitions to include this flag because we bring in uAMQP headers which depend on this flag :(.